### ISSUE TITLE
decode for generating only one response?

### ISSUE THREAD

**Issue by luomuqinghan** [hsgodhia/hred#3]
Thank you for your code!
But in original HRED, for k context and one response, HRED generates k utterances. 
It seems that you only generate the final response in training process?

**Comment by hsgodhia** [comment]
hi @luomuqinghan yes currently one the final utterance is decode, after reading the paper it seemed like that may be the right approach? Do you think every utterance along the way must be decoded based on the previous history?

**Comment by luomuqinghan** [comment]
Yes, it seems that the paper of "HRED"  did in this way when training. In test, it did in your way.
